node/src/accountant/mod.rs

```diff
@@ -1778,27 +1205,83 @@ mod tests {
                 pending_payable: vec![fingerprint],
                 response_skeleton_opt: Some(ResponseSkeleton {
                     client_id: 1234,
-                    context_id: 4321
+                    context_id: 4321,
                 }),
             }
         );
     }
 
+    #[test]
+    fn scan_request_from_ui_is_handled_in_case_the_scan_is_already_running() {
+        init_test_logging();
+        let test_name = "scan_request_from_ui_is_handled_in_case_the_scan_is_already_running";
+        let mut config = bc_from_earning_wallet(make_wallet("some_wallet_address"));
+        config.suppress_initial_scans_opt = Some(true);
+        config.scan_intervals_opt = Some(ScanIntervals {
+            payable_scan_interval: Duration::from_millis(10_000),
+            receivable_scan_interval: Duration::from_millis(10_000),
+            pending_payable_scan_interval: Duration::from_secs(100),
+        });
+        let fingerprint = PendingPayableFingerprint {
+            rowid_opt: Some(1234),
+            timestamp: SystemTime::now(),
+            hash: Default::default(),
+            attempt_opt: Some(1),
+            amount: 1_000_000,
+            process_error: None,
+        };
+        let pending_payable_dao =
+            PendingPayableDaoMock::default().return_all_fingerprints_result(vec![fingerprint]);
+        let mut subject = AccountantBuilder::default()
+            .bootstrapper_config(config)
+            .pending_payable_dao(PendingPayableDaoMock::new()) // For Accountant
+            .pending_payable_dao(PendingPayableDaoMock::new()) // For Payable Scanner
+            .pending_payable_dao(pending_payable_dao) // For PendingPayable Scanner
+            .build();
+        subject.logger = Logger::new(test_name);
+        let (blockchain_bridge, _, blockchain_bridge_recording_arc) = make_recorder();
+        let subject_addr = subject.start();
+        let system = System::new("test");
+        let peer_actors = peer_actors_builder()
+            .blockchain_bridge(blockchain_bridge)
+            .build();
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+        let ui_message = NodeFromUiMessage {
```

Comment:
It's usually slightly clearer if the try_send() calls are done together, which you can enable by shifting all the variables needed for those calls before the line where the very first try_send() call starts. Would be more becoming like that ;)

You can do the same again lower here.